### ISSUE TITLE
Bulk.insert

### ISSUE THREAD

**Issue by doubletaketech** [mikowals/batch-insert#2]
Does this package use Mongo's Bulk.insert()
http://docs.mongodb.org/manual/reference/method/Bulk.insert/#Bulk.insert
or does it loop through and make separate inserts for each doc?

I see CPU hangs when looping/inserting, so I'm looking for a better solution that does a bulk insert all at once instead of multiple insert operations.

Thanks
Don

**Comment by mikowals** [comment]
It loops inserts on the client to minimongo but on the server just calls the Node driver's insert method which takes an array.  I think internally Mongo handles the array with bulkInsert but I don't know for sure.

Your CPU problems might come from oplog tailing as meteor tries to keep clients in sync with your inserts.  See the warning note in the readme for this repo.

**Comment by doubletaketech** [comment]
Great!  Node driver will be fine.

How many docs have you tested?  Will 10K be an issue on the minimongo side?

**Comment by doubletaketech** [comment]
nevermind.  I see that you can run it only on the server, which is best in my case.

**Comment by doubletaketech** [comment]
Since your package uses node for the bulk insert, do you still see the oplog/cpu bug? 
I want the bulk insert to be done and not have meteor try to sync every insert to all clients as they happen. It should not sync to clients until the whole batch insert is complete.  Is that the case?

**Comment by mikowals** [comment]
This package doesn't impact these CPU effects related to observers though clearly they limit the usefulness of bulk operations.  All Mongo CRUD operations go through the oplog, no matter what the source, so replica sets can stay in sync.  You'll note that the [issue I referenced](https://github.com/meteor/meteor/issues/2668) has been closed with an attempted fix.  Some are still [having problems though](https://forums.meteor.com/t/oplog-tailing-too-far-behind-not-helping/2235/9).

If it is effecting you the first step would be trying to reduce the number of observers you have running for the changing data and next would be trying different settings for METEOR_OPLOG_TOO_FAR_BEHIND.  

In answer to your question of number of inserts, I have tried batch inserts of 10k and saw no problems with either Mongo or minimongo.  Internally Mongo [splits large batches into groups of 1000](http://docs.mongodb.org/v2.6/core/bulk-write-operations/#bulk-execution-mechanics).
